Port Troubleshooting Section from Screensteps (#131)
wpilibsuite/frc-docs: source/docs/networking/windows-firewall-configuration.rst

Windows Firewall Configuration
==============================

> Many of the programming tools used in FRC need network access for various reasons. Depending on the exact configuration, the Windows Firewall may potentially interfere with this access for one or more of these programs. This document describes procedures for Windows 7, but Windows 8 should be similar.

Reviewer: ```suggestion
Many of the programming tools used in FRC need network access for various reasons. Depending on the exact configuration, the Windows Firewall may potentially interfere with this access for one or more of these programs. This document describes procedures for Windows 7, but future versions should be similar.
```

Disabling Windows Firewall
--------------------------

The easiest solution is to disable the Windows Firewall. Teams should beware that this does make the PC potentially more vulnerable to malware attacks if connecting to the internet.

Control Panel
^^^^^^^^^^^^^

.. image:: images/windows-firewall-configuration-1.png

> Click ``Start->Control Panel`` to open the Control Panel. Click the dropdown next to ``View by:`` and select ``Small icons`` then click ``Windows Firewall``.

Reviewer 2: ```suggestion
Click ``Start`` -> ``Control Panel`` to open the Control Panel. Click the dropdown next to ``View by:`` and select ``Small icons`` then click ``Windows Defender Firewall``.
```

Turn Windows Firewall on or off
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. image:: images/windows-firewall-configuration-2.png

> In the left pane, click ``Turn Windows Firewall on or off``, and click yes or enter your Administrator password if a dialog appears.

Reviewer 2: ```suggestion
In the left pane, click ``Turn Windows Defender Firewall on or off``, and click yes. Enter your Administrator password if a dialog appears.
```

Disable the Firewall
^^^^^^^^^^^^^^^^^^^^

.. image:: images/windows-firewall-configuration-3.png

> For each category, select the radio button to ``Turn off Windows Firewall``. Then click OK.

Reviewer 2: ```suggestion
For each category, select the radio button to ``Turn off Windows Defender Firewall``. Then click OK.
```

Configure the firewall
----------------------

Alternatively, you can add exceptions to the Firewall for any FRC programs you are having issues with.

Open Control Panel
^^^^^^^^^^^^^^^^^^

.. image:: images/windows-firewall-configuration-4.png

> Click ``Start->Control Panel`` to open the Control Panel. Click the dropdown next to ``View by:`` and select ``Small icons`` then click ``Windows Firewall``.

Reviewer 2: ```suggestion
Click ``Start`` -> ``Control Panel`` to open the Control Panel. Click the dropdown next to ``View by:`` and select ``Small icons`` then click ``Windows Defender Firewall``.
```

Allow a program...
^^^^^^^^^^^^^^^^^^

.. image:: images/windows-firewall-configuration-5.png

> In the left pane, click ``Allow a program or feature through Windows Firewall``

Reviewer 2: ```suggestion
In the left pane, click ``Allow a program or feature through Windows Defender Firewall``
```
